lib/teleterm/apiserver/handler/handler_unified_resources.go

```diff
@@ -0,0 +1,101 @@
+// Copyright 2023 Gravitational, Inc
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package handler
+
+import (
+	"context"
+	"golang.org/x/exp/slices"
+
+	"github.com/gravitational/trace"
+
+	"github.com/gravitational/teleport/api/client/proto"
+	"github.com/gravitational/teleport/api/types"
+	api "github.com/gravitational/teleport/gen/proto/go/teleport/lib/teleterm/v1"
+	"github.com/gravitational/teleport/lib/client"
+	"github.com/gravitational/teleport/lib/teleterm/api/uri"
+)
+
+var supportedResourceKinds = []string{
+	types.KindNode,
+	types.KindDatabase,
+	types.KindKubernetesCluster,
+}
+
+func (s *Handler) ListUnifiedResources(ctx context.Context, req *api.ListUnifiedResourcesRequest) (*api.ListUnifiedResourcesResponse, error) {
+	clusterURI, err := uri.Parse(req.GetClusterUri())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	sortBy := types.SortBy{}
+	if req.GetSortBy() != nil {
+		sortBy.IsDesc = req.GetSortBy().IsDesc
+		sortBy.Field = req.GetSortBy().Field
+	}
```

Comment:
I was going to say that constructing `sortBy` here is debatable, but if it's just merely converting from over-the-wire format to a more specific type we use, then maybe it's fine.

But now I see that the main purpose of it is to help with converting the whole struct of `api.ListUnifiedResourcesRequest` to `proto.ListUnifiedResourcesRequest`. `proto.ListUnifiedResourcesRequest` is not even directly used by the daemon or `unifiedresources.List`, it is passed to a struct from outside of lib/teleterm.

As such, I think it'd be okay to pass `api.ListUnifiedResourcesRequest` unmodified to `unifiedresources.List` and convert it to `proto.ListUnifiedResourcesRequest` only there.

Reply:
>is not even directly used by the daemon or unifiedresources.List, it is passed to a struct from outside of lib/teleterm.

Yeah, that's true.
I have done it this way because I wanted to avoid passing down both the parsed `clusterURI` and the original `clusterURI` in the request struct. It seemed to me that it would be nice to prepare the data in the handler layer and give the deamon only the data it needs. 
But I see the problems with that, for example some logic is kept in the handler, some in the service.
Are we fine with having `clusterURI` in both places? If yes then I will address that separately.

Reply:
Ah, sorry, it's this kind of problem, I completely missed it. 🤦

idk how to deal with this in Go as there's no ergonomic way to say "This struct has the same fields as that other struct, except that Foo has type Bar and not Baz". Let's keep it as it is, I'll add this problem to my notes and later ask someone about it.